xclim/indices/_agro.py

```diff
@@ -589,6 +595,225 @@ def water_budget(
+        )
+
+
+    References
+    ----------
+    McKee, Thomas B., Nolan J. Doesken, and John Kleist. "The relationship of drought frequency and duration to time scales." In Proceedings of the 8th Conference on Applied Climatology, vol. 17, no. 22, pp. 179-183. 1993.
+    """
+    # "WPM" method doesn't seem to work for gamma or pearson3
+    dist_and_methods = {"gamma": ["ML", "APP"]}
+    if dist not in dist_and_methods.keys():
+        raise NotImplementedError(f"The distribution `{dist}` is not supported.")
+    elif method not in dist_and_methods[dist]:
+        raise NotImplementedError(
+            f"The method `{method}` is not supported for distribution `{dist}`."
+        )
+
+    # Determine group type
+    if freq == "D" or freq is None:
+        freq = "D"
+        group = "time.dayofyear"
+    else:
+        _, base, _, _ = parse_offset(freq)
+        if base in ["Y", "A", "Q", "M"]:
+            group = "time.month"
+        else:
+            raise NotImplementedError(f"Resampling frequency `{freq}` not supported.")
+
+    # Resampling precipitations
+    if freq != "D":
+        pr = pr.resample(time=freq).mean(keep_attrs=True)
+        pr_cal = pr_cal.resample(time=freq).mean(keep_attrs=True)
+
+        def needs_rechunking(da):
+            if uses_dask(da) and len(da.chunks[da.get_axis_num("time")]) > 1:
+                warnings.warn(
+                    "The input data is chunked on time dimension and must be fully rechunked to"
+                    " run `fit` on groups ."
+                    " Beware, this operation can significantly increase the number of tasks dask"
+                    " has to handle.",
+                    stacklevel=2,
+                )
+                return True
+            else:
+                return False
+
+        if needs_rechunking(pr):
+            pr = pr.chunk({"time": -1})
+        if needs_rechunking(pr_cal):
+            pr_cal = pr_cal.chunk({"time": -1})
+
+    # Rolling precipitations
+    if window > 1:
+        pr = pr.rolling(time=window).mean(skipna=False, keep_attrs=True)
+        pr_cal = pr_cal.rolling(time=window).mean(skipna=False, keep_attrs=True)
+
+    # Obtain fitting params and expand along time dimension
+    def resample_to_time(da, da_ref):
+        if freq == "D":
+            da = resample_doy(da, da_ref)
+        else:
+            da = da.rename(month="time").reindex(time=da_ref.time.dt.month)
+            da["time"] = da_ref.time
+        return da
+
+    params = pr_cal.groupby(group).map(lambda x: fit(x, dist, method))
+    params = resample_to_time(params, pr)
+
+    # ppf to cdf
+    # zero-bounded distributions;  'pearson3' will also go in this group once it's implemented
+    if dist in ["gamma", "pearson3"]:
+        prob_pos = dist_method("cdf", params, pr.where(pr > 0))
+        prob_zero = resample_to_time(
+            pr.groupby(group).map(
+                lambda x: (x == 0).sum() / x.dropna(dim="time").shape[0]
```

Comment:
```suggestion
                lambda x: (x == 0).sum('time') / x.notnull().sum('time')
```
We can't assume dimension order!